website/pages/en/cookbook/grafting-hotfix.mdx

```diff
@@ -0,0 +1,184 @@
+---
+Subgraph Best Practice 6 - Use Grafting for Quick Hotfix Deployment
+---
+
+## TLDR
+
+Grafting is a powerful feature in subgraph development that allows you to build and deploy new subgraphs while reusing the indexed data from existing ones. This technique is particularly useful for quickly deploying hotfixes to address critical issues without the need to re-index the entire subgraph from scratch. By preserving historical data, grafting ensures minimal downtime and maintains continuity in data services.
+
+## Why Grafting Is a Best Practice for Hotfixes
+
+1. **Rapid Deployment**
+
+   - **Minimize Downtime**: When a subgraph encounters a critical error and stops indexing, grafting enables you to deploy a fix immediately without waiting for re-indexing.
+   - **Immediate Recovery**: The new subgraph continues from the last indexed block, ensuring that data services remain uninterrupted.
+
+2. **Data Preservation**
+
+   - **Reuse Historical Data**: Grafting copies the existing data from the base subgraph, so you don’t lose valuable historical records.
+   - **Consistency**: Maintains data continuity, which is crucial for applications relying on consistent historical data.
+
+3. **Efficiency**
+   - **Save Time and Resources**: Avoids the computational overhead of re-indexing large datasets.
+   - **Focus on Fixes**: Allows developers to concentrate on resolving issues rather than managing data recovery.
+
+## Best Practices When Using Grafting for Hotfixes
+
+1. **Initial Deployment Without Grafting**
+
+   - **Start Clean**: Always deploy your initial subgraph without grafting to ensure that it’s stable and functions as expected.
+   - **Test Thoroughly**: Validate the subgraph’s performance to minimize the need for future hotfixes.
+
+2. **Implementing the Hotfix with Grafting**
+
+   - **Identify the Issue**: When a critical error occurs, determine the block number of the last successfully indexed event.
+   - **Create a New Subgraph**: Develop a new subgraph that includes the hotfix.
+   - **Configure Grafting**: Use grafting to copy data up to the identified block number from the failed subgraph.
+   - **Deploy Quickly**: Publish the grafted subgraph to restore service as soon as possible.
+
+3. **Post-Hotfix Actions**
+
+   - **Monitor Performance**: Ensure the grafted subgraph is indexing correctly and the hotfix resolves the issue.
+   - **Republish Without Grafting**: Once stable, deploy a new version of the subgraph without grafting for long-term maintenance.
+   - **Reason**: Relying on grafting indefinitely is not recommended as it can complicate future updates and maintenance.
```

Comment:
This could be a callout to give it more importance (if it warrants more attention)
```suggestion 
   > Note: Relying on grafting indefinitely is not recommended as it can complicate future updates and maintenance.
```